charts/steward/templates/clustertask-jenkinsfile-runner.yaml

```diff
@@ -70,10 +70,17 @@ spec:
       A textual description of the cause of this pipeline run. Will be set as cause of the Jenkins job.
       If null or empty, no cause information will be available.
     default: ""
+  - name: JFR_IMAGE
+    type: string
+    {{- with .Values.pipelineRuns.jenkinsfileRunner.image }}
+    default: {{ printf "%s:%s" .repository .tag | quote }}
+    {{- end }}
+    description: >
+      The jenkinsfile-runner image to be used.
   steps:
   - name: jenkinsfile-runner
     {{- with .Values.pipelineRuns.jenkinsfileRunner.image }}
-    image: {{ printf "%s:%s" .repository .tag | quote }}
+    image: $(inputs.params.JFR_IMAGE)
     imagePullPolicy: {{ .pullPolicy | quote }}
```

Comment:
__The image pull policy should be configurable from outside, too.__

For instance, if it is set to `Never` for the default image because the default image gets loaded by the image preloader, user-supplied images will never be pulled and pods hang in image pull backoff.

Another example: Users supply images with `latest` tag but default pull policy is `IfNotPresent`. Once the image exists on a node, it will not be updated anymore, which is not what users expect.

The image pull policy belongs to the configured image, so _don't mix defaults with user supplied values_:
- use _default_ image pull policy for _default_ image only
- use _user-supplied_ image pull policy (or `IfNotPresent`) for _user-supplied_ image

Reply:
How should this work? 

There is just simple variable substitution in the clustertask. See https://github.com/tektoncd/pipeline/blob/master/docs/variables.md

We cannot do a conditional substitution on this level.

For the pull policy we can only do the same as for the image. Then we get the default configured in the values.yaml.

Reply:
Implemented in the controller.
A manual creation of a task can still only overwrite the image and will then use the default policy.

Reply:
I propose _not_ to put the default image and pull policy into the Tekton TaskRun anymore. Instead put them into the pipeline runs config map and let the controller set them.

Reply:
Changed